### ISSUE TITLE
Names.find showing no results for some partial postcodes

### ISSUE THREAD

**Issue by sam596** [OrdnanceSurvey/osdatahub#94]
Not able to return any results for partial postcodes as shown below. Some postcodes work, some do not.

```python
names = NamesAPI(key)

names.find("AB22") #returns many

names.find("AB22 9") #returns none

names.find("AB22 9AA") #returns many
```

This does not match the API returns directly, if I go to `https://api.os.uk/search/names/v1/find?query=ab22+9&key=[my key]`, it still returns the desired, it just doesn't within Python.

**Comment by sam596** [comment]
Just as an update, following my own investigation.

`AB22 9` begins returning a load of rows in Python, but fails to finish, as it begins searching with the offset equal to the number of entries it has found, but there are no subsequent entries to find in the API.

For instance,

```first pass: AB22 9 returns 100 entries (1-100)

second pass: returns 100 entries (101-200)

third pass: returns 25 entries (201-225)

fourth pass: returns 0, http_error raised, no data returned from names.find("AB22 9")```

Error is around line 82 in `names_api.py`. I've temporarily fixed this by commenting out `raise_http_error(response)` replacing with a simple `pass`, as this is sufficient for my use case.

**Comment by JEPooley** [comment]
Hi Sam, thanks very much for getting in touch :) I've tried to implement a fix in version 1.2.7 - if a requests returns nothing but the status code is 200, it should bypass the Exception and return the correct number of features. I've added some test cases using AB22 and credited you in the CHANGELOG.md (hope that's okay - more than happy to remove if not!).

Hopefully the fix works, but please do give us a shout if not!
